components/properties/src/script.rs

```diff
@@ -10,11 +10,13 @@ use crate::props::Script;
 
 use icu_codepointtrie::{CodePointTrie, TrieValue};
 use icu_provider::yoke::{self, *};
+use icu_uniset::{UnicodeSet, UnicodeSetBuilder};
 use zerovec::{VarZeroVec, ZeroSlice};
 
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
+const CODE_POINT_MAX: u32 = 0x10ffff;
 const SCRIPT_X_SCRIPT_VAL: u16 = 0x03FF;
 const SCRIPT_VAL_LENGTH: u16 = 10;
```

Comment:
Nit/thought: There's a necessary relationship between the mask and the shift amount here. I've run into the same situation myself a few times in the casemapping code. It would be nice if we had better utilities / a more clear prescribed style for how to write this so that it's obvious that these are connected.

(This is just thinking out loud; I'm not asking for any specific changes here.)